### PR TITLE
Rename types

### PR DESCRIPTION
**Pull request by Yshayy**
Change all types to Agogos, switch endpoints and authprovider terminology with upstreams

### CODE REVIEW

File: services/graphql-server/extensions/upstreams/upstream.go [hunk 1]
```diff
@@ -47,18 +46,18 @@ func newEndpoint(epConfig *gqlconfig.GqlEndpoint) Endpoint {
...
-	endpoints = make(map[string]Endpoint)
+func Init(epConfigs []*gqlconfig.Upstream) {
+	upstreams = make(map[string]Upstream)
 	for _, epConfig := range epConfigs {
-		endpoints[epConfig.Host] = newEndpoint(epConfig)
+		upstreams[epConfig.Host] = newUpstream(epConfig)
 	}
 }
 
 // Get gets Endpoint by host
```
Comment (AleF83) [inline services/graphql-server/extensions/upstreams/upstream.go]:
```suggestion
// Get gets Upstream by host
```

File: services/graphql-server/extensions/upstreams/upstream.go [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
 import (
-	"net/http"
-
-	authproviders "agogos/extensions/authproviders"
+	upstreamAuthentications "agogos/extensions/upstreams/authentication"
 	gqlconfig "agogos/generated"
+	"net/http"
 )
 
 // Endpoint - interface for endpoint extension. Allows to change @http directive url or to add headers to http request created in resolver
```
Comment (AleF83) [inline services/graphql-server/extensions/upstreams/upstream.go]:
```suggestion
// Upstream - interface for upstream extension. Allows to change @http directive url or to add headers to http request created in resolver
```

File: services/graphql-server/extensions/upstreams/authentication/upstream_authentication.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	for _, apConfig := range apConfigs {
+		_, ok := upstreamAuthentications[apConfig.AuthType]
+		if !ok {
+			upstreamAuthentications[apConfig.AuthType] = make(map[string]UpstreamAuthentication)
+		}
+		upstreamAuthentications[apConfig.AuthType][apConfig.Authority] = newUpstreamAuthentication(apConfig)
+	}
+}
+
+// Get gets Endpoint by host
```
Comment (AleF83) [inline services/graphql-server/extensions/upstreams/authentication/upstream_authentication.go]:
```suggestion
// Get gets UpstreamAuthentication by authType and authority
```

File: services/graphql-server/extensions/upstreams/authentication/upstream_authentication.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func newUpstreamAuthentication(apConfig *gqlconfig.UpstreamAuthCredentials) UpstreamAuthentication {
+	// TODO: add more generic approach with addition of new types
+	return newUpstreamClientCredentials(apConfig)
+}
+
+var upstreamAuthentications map[string](map[string]UpstreamAuthentication)
+
+// Init initializes endpoint repository by
```
Comment (AleF83) [inline services/graphql-server/extensions/upstreams/authentication/upstream_authentication.go]:
```suggestion
// Init initialises upstream repository by config
```

File: services/graphql-server/extensions/upstreams/authentication/upstream_authentication.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package authentication
+
+import (
+	"net/http"
+
+	gqlconfig "agogos/generated"
+)
+
+// AuthProvider interface for AuthProvider extension. Supplies auth details to HTTP request
```
Comment (AleF83) [inline services/graphql-server/extensions/upstreams/authentication/upstream_authentication.go]:
```suggestion
// UpstreamAuthentication interface for UpstreamAuthentication extension. Supplies auth details to HTTP request
```

File: services/registry/src/sync/sync-upstream-auth-credentials.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+import { map, shareReplay, distinctUntilChanged, filter } from "rxjs/operators";
+import * as R from "ramda";
+
+const syncUpstreamAuthCredentials$ = gqlObjects$.pipe(
+  map(x => x.upstreamAuthCredentials || {}),
+  distinctUntilChanged(R.equals),
+  shareReplay(1)
+);
+
+// TODO: Add auth providers validation
```
Comment (AleF83) [inline services/registry/src/sync/sync-upstream-auth-credentials.ts]:
```suggestion
// TODO: Add UpstreamAuthCredentials validation
```

File: services/registry/src/sync/sync-upstreams.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import { map, shareReplay, distinctUntilChanged, filter } from "rxjs/operators";
+import * as R from "ramda";
+
+const syncUpstreams$ = gqlObjects$.pipe(
+  map(x => x.upstreams || {}),
+  distinctUntilChanged(R.equals),
+  shareReplay(1)
+);
+
+// TODO: add endpoint validation
```
Comment (AleF83) [inline services/registry/src/sync/sync-upstreams.ts]:
```suggestion
// TODO: add Upstream validation
```

**Comment by Yshayy** [comment]
UpstreamClientCredentials feels a bit too long, any idea for a better name?

**Comment by AleF83** [comment]
> UpstreamClientCredentials feels a bit too long, any idea for a better name?
ClientCredentials is the auth type so we need to keep it as a part of type name. So I don't think will be something much better...

**Pull request merged**